pkg/scheduler/plugins/elasticquota/core/quota_info.go

```diff
@@ -0,0 +1,135 @@
+	"sync"
+
+	"k8s.io/apimachinery/pkg/api/resource"
+
+	v1 "k8s.io/api/core/v1"
+	quotav1 "k8s.io/apiserver/pkg/quota/v1"
+	"sigs.k8s.io/scheduler-plugins/pkg/apis/scheduling/v1alpha1"
+)
+
+type QuotaCalculateInfo struct {
+	// The semantics of "max" is the quota group's upper limit of resources.
+	Max v1.ResourceList `json:"max"`
+	// The semantics of "min" is the quota group's guaranteed resources, if quota group's "request" less than or
+	// equal to "min", the quota group can obtain equivalent resources to the "request"
+	OriginalMin v1.ResourceList `json:"original_min"`
+	// If Child's sumMin is larger than totalResource, the value of OriginalMin should be scaled in equal proportion
+	//to ensure the correctness and fairness of min
+	AutoScaleMin v1.ResourceList `json:"auto_scale_min"`
+	// All assigned pods used
+	Used v1.ResourceList `json:"used"`
+	// All pods request
+	Request v1.ResourceList `json:"request"`
+	// SharedWeight determines the ability of quota groups to compete for shared resources
+	SharedWeight v1.ResourceList `json:"shared_weight"`
+	// Runtime is the current actual resource that can be used by the quota group
+	Runtime v1.ResourceList `json:"runtime"`
+}
+
+type QuotaInfo struct {
+	// QuotaName
+	Name QuotaNameType `json:"name"`
+	// Quota's ParentName
+	ParentName QuotaNameType `json:"parent_name"`
+	//IsParent quota group
+	IsParent bool `json:"is_parent"`
+	// If runtimeVersion not equal to quotaTree runtimeVersion, means runtime has been updated.
+	RuntimeVersion int64 `json:"runtimeVersion,omitempty"`
+	// Allow lent resource to other quota group
+	AllowLentResource bool               `json:"min_quota_share"`
+	CalculateInfo     QuotaCalculateInfo `json:"calculateInfo,omitempty"`
+	lock              sync.Mutex
+}
+
+func NewQuotaInfo(isParent, allowLentResource bool, name, parentName QuotaNameType) *QuotaInfo {
+	return &QuotaInfo{
+		Name:              name,
+		ParentName:        parentName,
+		IsParent:          isParent,
+		AllowLentResource: allowLentResource,
+		RuntimeVersion:    0,
+		CalculateInfo: QuotaCalculateInfo{
+			Max:          v1.ResourceList{},
+			AutoScaleMin: v1.ResourceList{},
+			OriginalMin:  v1.ResourceList{},
+			Used:         v1.ResourceList{},
+			Request:      v1.ResourceList{},
+			SharedWeight: v1.ResourceList{},
+			Runtime:      v1.ResourceList{},
+		},
+	}
+}
+
+// GetLimitRequestNoLock returns the min value of request and max, as max is the quotaGroup's upper limit of resources.
+// As the multi-hierarchy quota Model described in the PR, when passing a request upwards, passing a request exceeding its
+// max will result in a wrong/invalid runtime distribution. For example, parentQuotaGroup's Max is 20, childGroup's Max
+// is 10, and the childGroup's request is 30. If the child passes 30 request upwards and get a 20 runtime back
+//(limited by the parent's max is 20), the child can only use 10 (limited by its max).
+func (qi *QuotaInfo) GetLimitRequestNoLock() v1.ResourceList {
```

Comment:
I noticed that many functions have the suffix `NoLock` and are exported in uppercase. But the lock field of QuotaInfo is lowercase hidden, so I think only functions in the package can get the lock and need to call these functions. So these functions don't need to be exported in uppercase?